cfg_mgmt/rotate.py

```diff
@@ -57,53 +53,44 @@ def delete_cfg_element(
         logger.warning(
             f'{type_name} is not (yet) supported for automated deletion'
         )
-        return
+        return False
 
     try:
-        delete_func(cfg_element=cfg_element, cfg_dir=cfg_dir, cfg_queue_entry=cfg_queue_entry)
+        delete_func(
+            cfg_element=cfg_element,
+            cfg_factory=cfg_factory,
+            cfg_queue_entry=cfg_queue_entry,
+        )
     except Exception as e:
         logger.error(
             f"error deleting secret for cfg-type '{type_name}' with name "
             f"'{cfg_queue_entry.target.name}': {e}."
         )
         raise
 
-    new_cfg_queue: typing.List[cmm.CfgQueueEntry] = [
-        entry for entry in cfg_metadata.queue
-        if not entry == cfg_queue_entry
-    ]
-    write_config_queue(
-        cfg_queue=new_cfg_queue,
-        cfg_queue_file_path=os.path.join(cfg_dir, cmm.cfg_queue_fname),
-    )
-    git_helper.add_and_commit(
-        message=f'process config queue for {cfg_element._type_name}/{cfg_element.name()}',
-    )
-    try:
-        git_helper.push('@', target_ref)
-    except:
-        logger.warning('failed to push processed config queue - reverting')
-        git_helper.repo.git.reset('--hard', '@~')
+    return True
 
 
 def rotate_cfg_element(
-    cfg_dir: str,
     cfg_element: model.NamedModelElement,
-    target_ref: str,
-    github_cfg: model.github.GithubConfig,
-    cfg_metadata: cmm.CfgMetadata,
-    github_repo_path: str,
-) -> bool:
+    cfg_factory: model.ConfigFactory,
+) -> typing.Union[typing.Tuple[cfg_mgmt.revert_function, dict, model.NamedModelElement], None]:
```

Comment:
This differs from what we discussed. Since the values returned by this functions are actually needed, I've opted to switch from using `True`/`False` (and Errors) to using truthy/falsy values.
Not sure if `None` or an empty tuple are the best choice to signal noop.